ArkEcosystem/laravel-ui
# refactor: limit html options in markdown parser

## Summary

Related to https://app.clickup.com/t/p1bv2p and https://app.clickup.com/t/13kr2ar
Use with https://github.com/ArkEcosystem/marketsquare.io/pull/2164

***New** now also depends on https://github.com/ArkEcosystem/laravel-commonmark/pull/19

- Adds the feature for adding a "don't remind me again" checkbox when opening an external
- Ensures the markdown handles that external link component
- Admits some basic HTML tags like `string`, `ul`, etc. (basically the ones that can be added with markdown)
- Normalizes both HTML and markdown tags to ensure it looks great 
- Prevents XSS and errors
- Adds test coverage

### To test:

Merge this on [this PR on marketsquare](https://github.com/ArkEcosystem/marketsquare.io/pull/2164) and try to add external links, the HTML tags, scripts, etc

Basically, try to broke the editor, the app or add scripts and ensure the content you add looks as expected once rendered

🌮 🌮  if you find a vulnerability

## Checklist
